client/src/systems/shooter.rs

```diff
@@ -0,0 +1,83 @@
+use derive_new::new;
+use westiny_common::network::ShotEvent;
+use amethyst::core::math::{Point2, Vector2};
+use amethyst::prelude::{Builder, World};
+use amethyst::core::{Transform, Time};
+use std::time::Duration;
+use amethyst::renderer::SpriteRender;
+use crate::resources::SpriteResource;
+use westiny_common::resources::{SpriteId, AudioQueue, SoundId};
+use westiny_common::entities::spawn_bullet;
+use amethyst::core::ecs::{System, ReaderId, Read, LazyUpdate, Entities, SystemData, WriteExpect};
+use amethyst::core::ecs::shrev::EventChannel;
+
+#[derive(Default)]
+pub struct ShooterSystemDesc;
+
+impl<'a, 'b> ::amethyst::core::SystemDesc<'a, 'b, ShooterSystem> for ShooterSystemDesc {
+    fn build(self, world: &mut World) -> ShooterSystem {
+        <ShooterSystem as System<'_>>::SystemData::setup(world);
+
+        let reader_id = world
+            .fetch_mut::<EventChannel<ShotEvent>>()
+            .register_reader();
+
+        let bullet_sprite = world.fetch::<SpriteResource>().sprite_render_for(SpriteId::Bullet);
+        ShooterSystem::new(reader_id, bullet_sprite)
+    }
+}
+
+#[derive(new)]
+pub struct ShooterSystem {
+    shot_reader: ReaderId<ShotEvent>,
+    bullet_sprite: SpriteRender,
+}
+
+impl<'s> System<'s> for ShooterSystem {
+    type SystemData = (
+        Read<'s, EventChannel<ShotEvent>>,
+        Read<'s, Time>,
+        Read<'s, LazyUpdate>,
+        Entities<'s>,
+        WriteExpect<'s, AudioQueue>,
+    );
+
+    fn run(&mut self, (shot_event_channel, time, lazy, entities, mut audio): Self::SystemData) {
+        let current_time = time.absolute_time();
+        let events = shot_event_channel.read(&mut self.shot_reader);
+        if events.len() > 0 {
```

Comment:
current_time is irrelevant if there is no event (I general this system should not do anything if event queue is empty)
I don't know if it is idiomatic rust or not, but I would consider something like:
`if events.is_empty() {
  return;
}`